src/main/java/com/google/cloud/flink/bigquery/BigQueryTableSink.java

```diff
@@ -0,0 +1,261 @@
+  }
+
+  private void createBigQueryConfig() {
+    ImmutableMap<String, String> defaultOptions = ImmutableMap.of("bqWrite", "bqWrite");
+    this.allOptions.set(TABLE, table);
+    bqconfig =
+        FlinkBigQueryConfig.from(
+            requiredOptions(),
+            optionalOptions(),
+            allOptions,
+            defaultOptions,
+            new org.apache.hadoop.conf.Configuration(),
+            allOptions.get(DEFAULT_PARALLELISM),
+            new org.apache.flink.configuration.Configuration(),
+            allOptions.get(FLINK_VERSION),
+            Optional.empty());
+    BigQueryCredentialsSupplier bigQueryCredentialsSupplier =
+        new BigQueryCredentialsSupplier(
+            bqconfig.getAccessToken(),
+            bqconfig.getCredentialsKey(),
+            bqconfig.getCredentialsFile(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty());
+
+    FlinkBigQueryConnectorUserAgentProvider agentProvider =
+        new FlinkBigQueryConnectorUserAgentProvider(allOptions.get(FLINK_VERSION));
+    UserAgentHeaderProvider userAgentHeaderProvider =
+        new UserAgentHeaderProvider(agentProvider.getUserAgent());
+    this.bigQueryWriteClientFactory =
+        new BigQueryClientFactory(bigQueryCredentialsSupplier, userAgentHeaderProvider, bqconfig);
+  }
+
+  private void createBigQueryTable() throws JSQLParserException {
+    TableId tableId = bqconfig.getTableId();
+    if (bigQueryClient == null) {
+      final BigQuery bigquery =
+          BigQueryOptions.newBuilder()
+              .setCredentials(bqconfig.createCredentials())
+              .build()
+              .getService();
+      bigQueryClient =
+          new BigQueryClient(
+              bigquery, Optional.of(tableId.getProject()), Optional.of(tableId.getDataset()));
+    }
+    boolean destTableExists = bigQueryClient.tableExists(tableId);
+    if (!destTableExists) {
+      List<DataType> columnDataTypeList = Arrays.asList(fieldTypes);
+      List<String> columnNameList = Arrays.asList(fieldNames);
+      List<Field> listOfFields = new ArrayList<Field>();
+      for (int i = 0; i < columnNameList.size(); i++) {
+        RowField rowField =
+            new RowField(
+                columnNameList.get(i).toString(), columnDataTypeList.get(i).getLogicalType());
+        listOfFields.addAll(getFields(rowField, null));
+      }
+      FieldList fieldlist = FieldList.of(listOfFields);
+      Schema schema = Schema.of(fieldlist);
+      bigQueryClient.createTable(tableId, schema);
+    }
+  }
+
+  @Override
+  public TableSink<Row> configure(String[] fieldNames, TypeInformation[] fieldTypes) {
+    return null;
+  }
+
+  @Override
+  public DataType getConsumedDataType() {
+    return getTableSchema().toRowDataType();
+  }
+
+  @Override
+  public TableSchema getTableSchema() {
+    return TableSchema.builder().fields(fieldNames, fieldTypes).build();
+  }
+
+  @Override
+  public void setOverwrite(boolean overwrite) {}
```

Comment:
Missing implementation - how overwrite is supported?

Reply:
Implementing OverwritableTableSink class it self supports override and setOverwrite() is for setting the boolean value of overwrite option for using it for controlling some other logic if required. Currently we are not using to overwrite, So simply leaving this method with out any logic